### PR TITLE
improve the UI style customization as a follow-up of #2660

### LINKED ISSUE #2603: Navigation UI requires extending style definition
Currently, there are a bunch of attribute references in the UI SDK, like:
https://github.com/mapbox/mapbox-navigation-android/blob/1a288de10a0d6615adbd97be7c1ad60273a7f9bf/libnavigation-ui/src/main/res/values/attrs.xml#L63-L64

This means, that all of those references need to be filled in by a style object. This is done for example here:
https://github.com/mapbox/mapbox-navigation-android/blob/1a288de10a0d6615adbd97be7c1ad60273a7f9bf/libnavigation-ui/src/main/res/values/styles.xml#L42

However, this means that the application that uses Nav UI SDK has to use the default style, like `NavigationViewLight`, to fill in those references. This happens automatically for the `NavigationView`, but if a developer uses only standalone components like `NavigationMapboxMap`, the app will crash when the object is first instantiated:
```
android.content.res.Resources$NotFoundException: Resource ID #0x0
    at android.content.res.ResourcesImpl.getValueForDensity(ResourcesImpl.java:246)
    at android.content.res.Resources.getDrawableForDensity(Resources.java:905)
    at android.content.res.Resources.getDrawable(Resources.java:845)
    at android.content.Context.getDrawable(Context.java:687)
    at androidx.core.content.ContextCompat.getDrawable(ContextCompat.java:454)
    at com.mapbox.navigation.ui.ThemeSwitcher.retrieveThemeMapMarker(ThemeSwitcher.java:52)
    at com.mapbox.navigation.ui.map.NavigationMapboxMap.initializeNavigationSymbolManager(NavigationMapboxMap.java:699)
    at com.mapbox.navigation.ui.map.NavigationMapboxMap.<init>(NavigationMapboxMap.java:117)
    at com.mapbox.navigation.ui.map.NavigationMapboxMap.<init>(NavigationMapboxMap.java:99)
```

The only option is to extend the style `NavigationViewLight` and apply it as the whole app's theme, or blindly fill in missing resources. I think the SDK should provide the default resources out of the box, without having to implement the style, and allow overriding them in custom styles instead.

/cc @mapbox/navigation-android

### PR DESCRIPTION
**Pull request by JunDai**
## Description

Supplement of #2660 to better fix #2603.
As @Guardiola31337 noticed, after #2660 merged, all the UI SDK components in the example app shows the customized style which is not good.
Better to only show the customized style in the `CustomUIComponentStyleActivity` and all the other example still use the Mapbox default style. This PR addresses this issue.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Only show the customized style in the `CustomUIComponentStyleActivity` and all the other example still use the Mapbox default style.

### Implementation

* The UI SDK components support programmatically update View style after been initialized.
* The `example` app uses a different style name than the default Mapbox style name. So it only impact the `CustomUIComponentStyleActivity`

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt [hunk 1]
```diff
@@ -121,8 +121,9 @@ class NavigationViewActivity : AppCompatActivity(), OnNavigationReadyCallback, N
     }
 
     private fun getInitialCameraPosition(): CameraPosition {
+        val originCoordinate = route.routeOptions()?.coordinates()?.get(0)?.coordinates()
         return CameraPosition.Builder()
-            .target(LatLng(37.791503, -122.396255))
+            .target(LatLng(originCoordinate!![1], originCoordinate[0]))
```
Comment (JunDai) [inline examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt:126]: @Guardiola31337 : this addresses the https://github.com/mapbox/mapbox-navigation-android/pull/2818#discussion_r412374755

File: examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt [hunk 2]
```diff
@@ -121,8 +121,9 @@ class NavigationViewActivity : AppCompatActivity(), OnNavigationReadyCallback, N
     }
 
     private fun getInitialCameraPosition(): CameraPosition {
+        val originCoordinate = route.routeOptions()?.coordinates()?.get(0)?.coordinates()
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt:124]:
```suggestion
        val originCoordinate = route.routeOptions()?.coordinates()?.get(0)
```

File: examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt (hunk 1 above)
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt:126]:
```suggestion
            .target(LatLng(originCoordinate!!.latitude(), originCoordinate.longitude()))
```

File: libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationButton.java [hunk 3]
```diff
@@ -27,4 +29,9 @@
    * Shows the button
    */
   void show();
+
+  /**
+   * Set customized style
+   */
+  void updateStyle(@StyleRes int styleRes);
```
Comment (Guardiola31337) [inline libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationButton.java:36]: For documentation purposes noting that this breaks SemVer cc @langsmith

**Comment by JunDai** [comment]
> When testing these changes run into the following leaks

I think this leak isn't related to this PR, looks like similar to #2822 ?
#2823 created for tracking.

**Pull request merged**